src/triage/component/postmodeling/contrast/evaluator.py

```diff
@@ -0,0 +1,543 @@
+"""
+Model Evaluator
+
+This script contain a set of elements to help the postmodeling evaluation
+of audited models by triage.Audition. This will be a continuing list of
+routines that can be scaled up and grow according to the needs of the
+project, or other postmodeling approaches.
+
+To run most of this routines you will need:
+    - S3 credentials (if used) or specify the path to both feature and
+    prediction matrices.
+    - Working database db_engine.
+"""
+
+import pandas as pd
+import numpy as np
+from sqlalchemy.sql import text
+from matplotlib import pyplot as plt
+from sklearn import metrics
+from collections import namedtuple
+
+from utils.file_helpers import download_s3
+from utils.test_conn import db_engine
+from utils.aux_funcs import recombine_categorical
+
+# Get individual model_ids from Audition outcome
+
+
+def get_models_ids(audited_model_group_ids):
+    query = db_engine.execute(text("""
+    SELECT model_group_id,
+           model_id
+    FROM model_metadata.models
+    WHERE model_group_id = ANY(:ids);
+    """), ids=audited_model_group_ids)
+
+    ModelTuple = namedtuple('Model', ['model_group_id', 'model_id'])
+    l = [ModelTuple(*i) for i in query]
+
+    return(l)
+
+# Get indivual model information/metadata from Audition output
+
+
+class Model(object):
+    '''
+    ModelExtractor class calls the model metadata from the database
+    and hold model_id metadata features on each of the class attibutes.
+    This class will contain any information about the model, and will be
+    used to make comparisons and calculations across models. 
+
+    A pair of (model_group_id, model_id) is needed to instate the class. These
+    can be feeded from the get models_ids. 
+    '''
+    def __init__(self, model_group_id, model_id):
+        self.model_id = model_id
+        self.model_group_id = model_group_id
+
+        # Retrive model_id metadata from the model_metadata schema
+        model_metadata = pd.read_sql("""
```

Comment:
Like the function/utility earlier in this module, could this not just execute a query against the engine? It doesn't _seem_ like pandas or a data frame are necessary.